### PR TITLE
fix(manager/gitlabci): clean more than one slash

### LINKED ISSUE #16600: [gitlab-ci] Filesystem access violation with .gitlab-ci.yml
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.116.0

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Since release `32.115.0` there's an error with `.gitlab-ci.yml` if `include` starts with double `/`.
Double `/` is a valid construction previously advertised as hack to https://gitlab.com/gitlab-org/gitlab/-/issues/28987.

This was introduced by #16511.

[Sample repo](https://gitlab.com/vitoyucepi/renovate-issue-16600).

### Relevant debug logs

<details><summary>Logs</summary>

```
 WARN: Preventing access to file outside the base directory (repository=vitoyucepi/renovate-issue-16600)
       "path": "/.gitlab-ci.yml",
       "baseDir": "/tmp/renovate/repos/gitlab/vitoyucepi/renovate-issue-16600"
ERROR: Repository has unknown error (repository=vitoyucepi/renovate-issue-16600)
       "err": {
         "message": "file-access-violation-error",
         "stack": "Error: file-access-violation-error\n    at assertBaseDir (/usr/src/app/node_modules/renovate/lib/util/fs/util.ts:12:11)\n    at ensurePath (/usr/src/app/node_modules/renovate/lib/util/fs/util.ts:21:3)\n    at ensureLocalPath (/usr/src/app/node_modules/renovate/lib/util/fs/util.ts:26:10)\n    at readLocalFile (/usr/src/app/node_modules/renovate/lib/util/fs/index.ts:34:40)\n    at Object.extractAllPackageFiles (/usr/src/app/node_modules/renovate/lib/modules/manager/gitlabci/extract.ts:130:40)\n    at extractAllPackageFiles (/usr/src/app/node_modules/renovate/lib/modules/manager/index.ts:49:17)\n    at getManagerPackageFiles (/usr/src/app/node_modules/renovate/lib/workers/repository/extract/manager-files.ts:33:29)\n    at /usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:47:28\n    at async Promise.all (index 0)\n    at extractAllDependencies (/usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:45:26)\n    at checkOnboardingBranch (/usr/src/app/node_modules/renovate/lib/workers/repository/onboarding/branch/index.ts:58:22)\n    at getRepoConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/config.ts:15:12)\n    at initRepo (/usr/src/app/node_modules/renovate/lib/workers/repository/init/index.ts:37:12)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:40:14)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:154:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by hasanwhitesource**
## Changes

Clear more than one slash from the include block in gitlabci.

## Context

According to https://gitlab.com/gitlab-org/gitlab/-/issues/28987#note_392419953 there can be more than one slash.
Currently renovate is giving "Preventing access to file outside the base directory" error.
If we take care of extra slashes we are able to detect the dependencies without and error :
Ran on: https://gitlab.com/hasan.awad/myfirsttestdata
image
Closes #16600

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] Both unit tests + ran on a real repository
- [x] No unit tests but ran on a real repository, or

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/gitlabci/extract.ts [hunk 1]
```diff
@@ -146,15 +146,15 @@ export async function extractAllPackageFiles(
     if (is.array(doc?.include)) {
       for (const includeObj of doc.include) {
         if (is.string(includeObj.local)) {
-          const fileObj = includeObj.local.replace(regEx(/^\//), '');
+          const fileObj = includeObj.local.replace(regEx(/^\/+/), '');
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:149]: use trim leading slash function from url util?

File: lib/modules/manager/gitlabci/extract.ts [hunk 2]
```diff
@@ -146,15 +146,15 @@ export async function extractAllPackageFiles(
...
           if (!seen.has(fileObj)) {
             seen.add(fileObj);
             filesToExamine.push(fileObj);
           }
         }
       }
     } else if (is.string(doc?.include)) {
-      const fileObj = doc.include.replace(regEx(/^\//), '');
+      const fileObj = doc.include.replace(regEx(/^\/+/), '');
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:157]: same here?

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/main/lib/util/url.ts#L30

should probably remove also multiple leading slashes

File: lib/modules/manager/gitlabci/extract.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/manager/gitlabci/extract.ts:149]: It only trims one :)

Comment (viceice) [reply lib/modules/manager/gitlabci/extract.ts:149]: i know, see my other comment, it should trim all, like the trailing variant

**Comment by hasanwhitesource** [comment]
@viceice don't think this will work since that only trims 1 slash

**Comment by viceice** [comment]
> Change this in the same pr ? 🤔

sure, it's a very small change and probably expected. If it fails too many tests then we should do in seperate PR

File: lib/modules/manager/gitlabci/extract.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/manager/gitlabci/extract.ts:149]: OK, so let's improve trimLeadingSlash to replace multiple and use that function instead.

**Review by rarkins — APPROVED** [review]

**Pull request merged**